lib/plugins/aws/invokeLocal/index.js

```diff
@@ -125,60 +128,75 @@ class AwsInvokeLocal {
-        this.options.data,
-        this.options.context);
-    }
 
-    if (_.includes(['python2.7', 'python3.6', 'python3.7'], runtime)) {
-      const handlerComponents = handler.split(/\./);
-      const handlerPath = handlerComponents.slice(0, -1).join('.');
-      const handlerName = handlerComponents.pop();
-      return this.invokeLocalPython(
-        process.platform === 'win32' ? 'python.exe' : runtime,
-        handlerPath,
-        handlerName,
-        this.options.data,
-        this.options.context);
-    }
-
-    if (runtime === 'java8') {
-      const className = handler.split('::')[0];
-      const handlerName = handler.split('::')[1] || 'handleRequest';
-      return this.invokeLocalJava(
-        'java',
-        className,
-        handlerName,
-        this.serverless.service.package.artifact,
-        this.options.data,
-        this.options.context);
-    }
+    const layers = _.mapKeys(
+      this.serverless.service.layers,
+      (value, key) => this.provider.naming.getLambdaLayerLogicalId(key)
+    );
+    const layerPathsPromise = Promise.all(
+      (this.options.functionObj.layers || this.serverless.service.provider.layers || [])
+        .map(layer => {
+          if (layer.Ref) {
+            return layers[layer.Ref].path;
+          }
+          const arnParts = layer.split(':');
+          const layerArn = arnParts.slice(0, -1).join(':');
+          const layerVersion = Number(arnParts.slice(-1)[0]);
+          this.serverless.cli.consoleLog(`Downloading layer ${layer}...`);
+          return this.provider.request(
+            'Lambda', 'getLayerVersion', { LayerName: layerArn, VersionNumber: layerVersion })
+            .then(layerInfo => download(layerInfo.Content.Location))
+            .then(jsZip.loadAsync)
+            .then(zip => Promise.all(
+              Object.keys(zip.files)
+                .map(filename => zip.files[filename].async('nodebuffer').then(fileData => {
+                  if (filename.endsWith('/')) {
+                    return Promise.resolve();
+                  }
+                  mkdirp.sync(
+                    path.dirname(`.serverless/invokeLocal/layers/${arnParts[6]}/${filename}`));
+                  return fs.writeFileAsync(
+                    `.serverless/invokeLocal/layers/${arnParts[6]}/${filename}`, fileData, {
+                      mode: zip.files[filename].unixPermissions,
+                    });
+                }))))
+            .then(() => `.serverless/invokeLocal/layers/${arnParts[6]}`);
+        }));
+
+    const imageName = 'sls-docker';
+
+    return layerPathsPromise.then(layerPaths => new BbPromise((resolve, reject) => {
+      let dockerfile = `FROM lambci/lambda:${runtime}`;
+      for (const layerPath of layerPaths) {
+        dockerfile += `\nADD --chown=sbx_user1051:495 ${layerPath} /opt`;
+        // fs.copySync(layerPath, `.serverless/invokeLocal/${layerPath}`);
+      }
+      mkdirp.sync(path.dirname('.serverless/invokeLocal'));
```

Comment:
I had to replace this with:

```suggestion
      mkdirp.sync(path.join('.serverless' , 'invokeLocal'));
```